File: plugins/OrchidPosts/src/main/java/com/eden/orchid/posts/pages/PostTagArchivePage.java
package com.eden.orchid.posts.pages;

import com.eden.common.util.EdenUtils;
import com.eden.orchid.api.resources.resource.OrchidResource;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter @Setter
public class PostTagArchivePage extends PostArchivePage {

    public PostTagArchivePage(OrchidResource resource) {
        super(resource, "postTagArchive");
        this.setTemplates(new String[]{"postTagArchive", "postArchive"});
    }

    public String getTag() {
        return category;
    }

    public void setTag(String tag) {
        this.category = tag;
    }

    @Override
    public List<String> getTemplates() {
        List<String> templates = super.getTemplates();
        if(!EdenUtils.isEmpty(category)) {
            templates.add(0, "postTagArchive-" + category);
        }

        return templates;
    }
}

File: plugins/OrchidPosts/src/main/kotlin/com/eden/orchid/posts/pages/PostTagArchivePage.kt
package com.eden.orchid.posts.pages

import com.eden.common.util.EdenUtils
import com.eden.orchid.api.options.annotations.Archetype
import com.eden.orchid.api.options.archetypes.ConfigArchetype
import com.eden.orchid.api.resources.resource.OrchidResource

@Archetype(value = ConfigArchetype::class, key = "postTagArchivePages")
class PostTagArchivePage(resource: OrchidResource, index: Int, permalink: String) : PostArchivePage(resource, "postTagArchive", index, permalink) {

    var tag: String
        get() = category!!
        set(tag) {
            this.category = tag
        }

    init {
        this.setTemplates(arrayOf("postTagArchive", "postArchive"))
    }

    override fun getTemplates(): List<String> {
        val templates = super.getTemplates().toMutableList()
        if (!EdenUtils.isEmpty(category)) {
            templates.add(0, "postTagArchive-" + category)
        }

        return templates
    }
}

